### PR TITLE
feat(mesh): login with Dynamic-xyz

### PR DESCRIPTION
**Pull request by tyc67**
## Notable Changes

- integrate with dynamic.xyz [sdk
](https://docs.dynamic.xyz/adding-dynamic/adding-the-sdk)
- integrate dynamic UI Component use [DynamicWidget](https://docs.dynamic.xyz/react-sdk/components/dynamicwidget) for interfaces.
  - configure `<DynamicContextProvider />` and customized dynamic UI according to [doc1](https://docs.dynamic.xyz/design-customizations/tutorials/login-views-guide#adding-to-the-dynamiccontextprovider), [doc2](https://docs.dynamic.xyz/react-sdk/providers/dynamiccontextprovider)
  - customized dynamicwidget button [doc](https://docs.dynamic.xyz/design-customizations/tutorials/design-tutorial#button-text-and-font)
- add `wallet` field to `GetCurrentUserMemberId` to check if the user is connected with Dynamic.
- if user has a dynamic wallet display #797 in `<MeshPoint />` , 
- if user does not have a dynamic wallet show  `<CreateMeshPoint />` [figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=8983-39607&t=0wM9MekCHcdvVAec-4)
- refactor login process `<LoginSetWallet/>`: On successful Dynamic sign-up, use the `updateMemberWallet` function to update the wallet in the CMS.
- temporarily hide the skip button in `<LoginStepsTitle />` for this ndx mid-term review.
- show a spinner during the loading state

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207793413604580

### CODE REVIEW

File: packages/mesh/constants/config.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ let FIREBASE_CONFIG = {
   APP_ID: '',
 }
 
+const DYNAMIC_ENV_ID = '51dfdb4d-d8ef-47e4-8a89-57ea9f1b3803'
```
Comment (erase2004) [inline packages/mesh/constants/config.ts]: DYNAMIC_ENV_ID 是所有環境都共用？

File: packages/mesh/app/layout.tsx [hunk 2]
```diff
@@ -19,7 +25,51 @@ export default function RootLayout({
...
+                    },
+                  ],
+                },
+              ],
+            },
+            cssOverrides:
+              '.wallet-list-item__tile { background-color: lightblue; }',
+            displaySiweStatement: true,
+            privacyPolicyUrl: 'https://www.dynamic.xyz/privacy-policy',
+            termsOfServiceUrl: 'https://www.dynamic.xyz/terms-of-service',
```
Comment (erase2004) [inline packages/mesh/app/layout.tsx:50]: `privacyPolicyUrl` 和 `termsOfServiceUrl` 應該加上 TODO comment，未來更新連結資訊？

File: packages/mesh/app/point/layout.tsx [hunk 3]
```diff
@@ -1,3 +1,5 @@
+'use client'
```
Comment (erase2004) [inline packages/mesh/app/point/layout.tsx]: 這裡會需要 `'use client'` 的原因？

File: packages/mesh/app/point/_components/create-mesh-point.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          innerButtonComponent={
+            <div className="flex flex-row items-center">
+              <span>以Dynamic繼續</span>
+            </div>
+          }
+        />
+      </div>
+      <p className="footnote text-primary-400">
+        讀選點數是什麼？
+        <Link href={'/'}>
```
Comment (erase2004) [inline packages/mesh/app/point/_components/create-mesh-point.tsx]: Add TODO comment

File: packages/mesh/app/point/_components/create-mesh-point.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    <div className="flex flex-col items-center gap-6 px-5 py-10 sm:px-10">
+      <p className="subtitle-1 text-center text-primary-700">
+        您尚未新增/連結錢包。點擊下方按鈕成功建立錢包，即可獲得 100 讀選點數！
+      </p>
+      <div className="flex w-full max-w-[320px] items-center justify-center">
+        <DynamicWidget
+          buttonClassName=""
+          buttonContainerClassName=""
+          innerButtonComponent={
+            <div className="flex flex-row items-center">
```
Comment (erase2004) [inline packages/mesh/app/point/_components/create-mesh-point.tsx]: Snipaste_2024-07-31_20-42-42
按鈕缺少 Icon

File: packages/mesh/app/point/_components/mesh-point.tsx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+                  // 'use server'
+                  //TODO: update fetching logic
+                  return transactionData
+                }}
+              />
+            )}
+          </section>
+        </>
+      ) : (
+        <div className="flex h-[calc(100vh-123px)] items-center justify-center sm:h-[calc(100vh-424px)]">
```
Comment (erase2004) [inline packages/mesh/app/point/_components/mesh-point.tsx]: 這個狀態似乎不在設計稿？

File: packages/mesh/app/login/_components/login-set-wallet.tsx [hunk 7]
```diff
@@ -1,27 +1,51 @@
...
-          color="white"
-          icon={{ iconName: 'icon-dynamicxyz', size: 'm' }}
-          text="以Dynamic繼續"
-          onClick={() => router.push('/point')}
+      <div className="flex w-full max-w-[320px] justify-center">
+        <DynamicWidget
+          buttonClassName=""
+          buttonContainerClassName=""
+          innerButtonComponent={
+            <div className="flex flex-row items-center">
```
Comment (erase2004) [inline packages/mesh/app/login/_components/login-set-wallet.tsx]: 似乎少了 Icon ?

**Comment by erase2004** [comment]
這個區塊在 768 以上應該是垂直置中？
Snipaste_2024-07-31_20-34-57

File: packages/mesh/constants/config.ts (hunk 1 above)
Comment (tyc67) [reply packages/mesh/constants/config.ts]: 修改於 [9285442](https://github.com/readr-media/Sachiel/pull/812/commits/92854426d95252b0eb354312e0b5431763cc4286)

File: packages/mesh/app/layout.tsx (hunk 2 above)
Comment (tyc67) [reply packages/mesh/app/layout.tsx:50]: 修改於 [9285442](https://github.com/readr-media/Sachiel/pull/812/commits/92854426d95252b0eb354312e0b5431763cc4286)

File: packages/mesh/app/point/layout.tsx (hunk 3 above)
Comment (tyc67) [reply packages/mesh/app/point/layout.tsx]: 已移除, 修改於[9285442](https://github.com/readr-media/Sachiel/pull/812/commits/92854426d95252b0eb354312e0b5431763cc4286)

File: packages/mesh/app/point/_components/create-mesh-point.tsx (hunk 4 above)
Comment (tyc67) [reply packages/mesh/app/point/_components/create-mesh-point.tsx]: 修改於 [9285442](https://github.com/readr-media/Sachiel/pull/812/commits/92854426d95252b0eb354312e0b5431763cc4286)

File: packages/mesh/app/point/_components/create-mesh-point.tsx (hunk 5 above)
Comment (tyc67) [reply packages/mesh/app/point/_components/create-mesh-point.tsx]: 補 Icon 於 [9285442](https://github.com/readr-media/Sachiel/pull/812/commits/92854426d95252b0eb354312e0b5431763cc4286)

File: packages/mesh/app/point/_components/mesh-point.tsx (hunk 6 above)
Comment (tyc67) [reply packages/mesh/app/point/_components/mesh-point.tsx]: 主要是用於dynamic 登入時效過後須重新登入的狀態, 有請設計出稿 [figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=9865-71040&t=hemMjOgLvdrXjdNH-4) , 修改於 [9285442](https://github.com/readr-media/Sachiel/pull/812/commits/92854426d95252b0eb354312e0b5431763cc4286)

File: packages/mesh/app/login/_components/login-set-wallet.tsx (hunk 7 above)
Comment (tyc67) [reply packages/mesh/app/login/_components/login-set-wallet.tsx]: 補 Icon 於 [9285442](https://github.com/readr-media/Sachiel/pull/812/commits/92854426d95252b0eb354312e0b5431763cc4286)

**Comment by tyc67** [comment]
> 這個區塊在 768 以上應該是垂直置中？ Snipaste_2024-07-31_20-34-57

修改於 [9285442](https://github.com/readr-media/Sachiel/pull/812/commits/92854426d95252b0eb354312e0b5431763cc4286) 

Screenshot 2024-08-01 at 4 10 47 PM

**Review by erase2004 — APPROVED** [review]
LGTM 👏

**Pull request merged**